docs/001_develop/03_client-capabilities/021_desktop-interoperability/002_fdc3-channel-event.mdx

```diff
@@ -0,0 +1,356 @@
+    return (
+        <rapid-grid-pro>
+            <grid-pro-genesis-datasource
+                resource-name="ALL_INSTRUMENT_REF">
+            </grid-pro-genesis-datasource>
+            <fdc3-channel-event
+                event-name="rowClicked"
+                channel-type="fdc3.position"
+                channel-name="appChannel">
+            </fdc3-channel-event>
+        </rapid-grid-pro>
+    );
+}
+```
+</TabItem>
+
+<TabItem value="angular">
+Declaration
+```html
+<fdc3-channel-event></fdc3-channel-event>
+```
+
+Usage
+```typescript
+import { Component, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+@Component({
+    selector: 'my-root',
+    templateUrl: './my.component.html',
+    standalone: true,
+    schemas: [CUSTOM_ELEMENTS_SCHEMA],
+    imports: [FormsModule],
+})
+export class AppComponent {}
+```
+
+View
+```html
+<rapid-grid-pro>
+    <grid-pro-genesis-datasource
+        resource-name="ALL_INSTRUMENT_REF">
+    </grid-pro-genesis-datasource>
+    <fdc3-channel-event
+        event-name="rowClicked"
+        channel-type="fdc3.position"
+        channel-name="appChannel">
+    </fdc3-channel-event>
+</rapid-grid-pro>
+```
+</TabItem>
+</Tabs>
+
+
+### Example 3 - Send mapped event on system channel
+
+This will listen to the `rowClicked` event that is fired by the `rapid-grid-pro` element whenever a row is clicked.
+
+It will broadcast on the current system color channel.
+
+It will broadcast with the context type `fdc3.position`.
+
+The mapping function will map the `event.detail.data` payload and create a new object.
+
+<Tabs defaultValue="genesis" values={[{ label: 'Genesis', value: 'genesis', }, { label: 'React', value: 'react', }, { label: 'Angular', value: 'angular', }]}>
+
+<TabItem value="genesis">
+Declaration
+```html
+<fdc3-channel-event></fdc3-channel-event>
+```
+Usage
+```typescript
+@customElement({
+    name: 'my-element',
+    template: html`
+        <rapid-grid-pro>
+            <grid-pro-genesis-datasource
+                resource-name="ALL_INSTRUMENT_REF">
+            </grid-pro-genesis-datasource>
+            <fdc3-channel-event
+                :mappingFunction="${() => ({TICKER, CUSIP}) => ({ id: { ticker: TICKER, cusip: CUSIP } })}"
+                event-name="rowClicked"
+                channel-type="fdc3.position">
+            </fdc3-channel-event>
+        </rapid-grid-pro>
+    `,
+})
+export class MyElement extends GenesisElement {}
+```
+</TabItem>
+
+<TabItem value="react">
+Declaration
+```html
+<fdc3-channel-event></fdc3-channel-event>
+```
+Usage
+```typescript
+export function MyComponent() {
+    return (
+        <rapid-grid-pro>
+            <grid-pro-genesis-datasource
+                resource-name="ALL_INSTRUMENT_REF>
+            </grid-pro-genesis-datasource>
+            <fdc3-channel-event
+                mappingFunction="{() => ({TICKER, CUSIP}) => ({ id: { ticker: TICKER, cusip: CUSIP } })}"
```

Comment:
Is this defo the correct syntax when doing it in react?

Reply:
Updated the react example and tested it